### PR TITLE
Remove react-router

### PR DESCRIPTION
**Pull request by nfour**
- [x] Removed `react-router`
- [x] Reconfigured react anchor scroller to include hash always, and for the demo to be able to override configs
- [x] `hash` is now passed around instead of the `location` object
- [x] Fix regression with hash causing navigation to follow

### CODE REVIEW

File: src/containers/Base/Base.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ export default class Base extends React.PureComponent {
   }
 
   render () {
-    const { location, classes } = this.props
+    const { hash = '', classes } = this.props
```
Comment (brendo) [inline src/containers/Base/Base.js]: Why did I not know this was possible!

**Review by brendo — CHANGES_REQUESTED** [review]
I noticed with this change that the app doesn't handle with the hash changes. 

For example:

1. http://localhost:8200/?url=https://temando.github.io/open-api-renderer/petstore-open-api-v3.0.0-RC2.json#/pet/{petId}/post
2. Notice the active state in the navigation for this method
3. Scroll further down, notice the hash in the URL updates, but the active state doesn't update with it. This is also noticeable if you scroll down into the store/user tags, the left navigation doesn't "follow" along, which is a regression on the current behaviour.

File: src/containers/Base/Base.js (hunk 1 above)
Comment (nfour) [reply src/containers/Base/Base.js]: haha

**Comment by nfour** [comment]
Will fix that :+1:

File: src/entries/demo/Demo.js [hunk 2]
```diff
@@ -2,13 +2,16 @@ import React from 'react'
...
 import Lincoln from '../Lincoln'
 import { styles } from './Demo.styles'
-import PropTypes from 'prop-types'
 
 jss.setup(preset())
+configureAnchors({ offset: -52, scrollDuration: 200, keepLastAnchorHash: true })
 
-const definitionUrl = parseQuery(location.search.split('?')[1]).url
+const definitionUrl = parseQuery(window.location.search.split('?')[1]).url
+const hash = window.location.hash
```
Comment (nfour) [inline src/entries/demo/Demo.js:14]: I suspect this isnt working

File: src/containers/Base/Base.js [hunk 3]
```diff
@@ -16,23 +19,46 @@ export default class Base extends React.PureComponent {
...
 
-    configureAnchors({ offset: -10, scrollDuration: 100 })
+    if (listenToHash) {
+      const history = inputHistory || createBrowserHistory()
+
+      this.setState({ history })
+
+      this.stopListeningToHistory = history.listen((location) => {
+        const { hash } = location
+        console.log(hash, location.hash)
```
Comment (phlorie) [inline src/containers/Base/Base.js]: `console.log` should be removed

Comment (nfour) [reply src/containers/Base/Base.js]: :ok_hand:

**Review by brendo — APPROVED** [review]

**Pull request merged**